agcaballero/portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -38,17 +38,25 @@
  */
 @WebServlet("/data")
 public class DataServlet extends HttpServlet {
-
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    // Retrieve max number of comments (if no info provided, default is 50)
+    int maxComments = Integer.parseInt(getParameter(request, "max-comments", "50"));
+
     // Retrieve from Datastore entities of type "Comment"
     DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
     Query query = new Query("Comment").addSort("timestamp", SortDirection.DESCENDING);
```

Comment:
Can a limit be set so that we don't fetch more data than is needed?
https://cloud.google.com/datastore/docs/concepts/queries#cursors_limits_and_offsets

If so it seems more optimal to fetch only what's needed rather than fetch all and then discard the unneeded portion.

Reply:
The link is to the Cloud API for Datastore not the AppEngine API. As such, I can't apply that specific fix (to my knowledge) as it uses a different version of the Query class. I agree that the current code is not optimal, and I would prefer to fetch less data. I used the FetchOptions class to somewhat improve this, but I believe I'm still fetching unnecessary data. I can't seem to find a similar function/class that caps the number of entities a query fetches within the AppEngine API.

Reply:
It seems using FetchOptions does fix this issue, so this thread is resolved.

Reply:
Done